### PR TITLE
[Better Tablet Products] Remove feature flag

### LINKED ISSUE #10958: [Better Tablet Products] Remove feature flag

### PR DESCRIPTION
**Pull request by kidinov**
Closes: #10958

### Description

Removes `BETTER_TABLETS_SUPPORT_PRODUCTS` FF

### Testing instructions

* Run the major Products List/Details flows on a phone and tablets

### Images/gif

- [x] I have considered if this change warrants release notes and have added them to `RELEASE-NOTES.txt` if necessary. Use the "[Internal]" label for non-user-facing changes.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">PR requires at least one label.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ProductListViewModel.kt [hunk 1]
```diff
@@ -447,13 +445,13 @@ class ProductListViewModel @Inject constructor(
...
 
     private fun openFirstLoadedProductOnTablet(products: List<Product>) {
-        if (products.isNotEmpty()) {
-            if (isTabletLogicNeeded() && openedProductId == null) {
-                openedProductId = products.first().remoteId
-                onOpenProduct(openedProductId!!, null)
-            }
-        } else {
-            if (isTabletLogicNeeded()) {
+        if (isTablet()) {
```
Comment (malinajirka) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ProductListViewModel.kt:448]: @kidinov Could you please clarify this piece of code for me. I'm not sure I follow the logic here - looking at https://github.com/woocommerce/woocommerce-android/pull/11021, it seems this PR changes that logic, is that intentional?

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/woocommerce/woocommerce-android/milestone/197">17.7</a>. This milestone is due in less than 2 days.<br />
Please make sure to get it merged by then or assign it to a milestone with a later deadline.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ProductListViewModel.kt (hunk 1 above)
Comment (kidinov) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ProductListViewModel.kt:448]: Sure! Before it was

```
        if (products.isNotEmpty()) {
            if (isTabletLogicNeeded() && openedProductId == null) {
                openedProductId = products.first().remoteId
                onOpenProduct(openedProductId!!, null)
            }
        } else {
            if (isTabletLogicNeeded()) {
                triggerEvent(ProductListEvent.OpenEmptyProduct)
            }
        }
```

Now

```
if (isTablet()) {
            if (products.isNotEmpty()) {
                if (openedProductId == null) {
                    openedProductId = products.first().remoteId
                    onOpenProduct(openedProductId!!, null)
                }
            } else {
                triggerEvent(ProductListEvent.OpenEmptyProduct)
            }
        }
```

Notice that both conditions could be executed only if `isTabletLogicNeeded` true. To simplify and make it easier to read I  lifted this up, so it's clear that all this code is tablet-specific

The change is intentional, yes

Comment (malinajirka) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ProductListViewModel.kt:448]: Thank you! I should have checked the code in AS - I overlooked which brackets are associated with each other. Now it all makes sense.

**Review by malinajirka — APPROVED** [review]
LGTM, thanks! (Some unit tests are failing thought)

**Pull request merged**